pkg/hintrunner/hinter/scope.go

```diff
@@ -65,36 +86,48 @@ func (sm *ScopeManager) AssignVariables(values map[string]any) error {
 	return nil
 }
 
-func (sm *ScopeManager) GetVariableValue(name string) (any, error) {
+func (sm *ScopeManager) GetScopeValue(name string) (*ScopeValue, error) {
 	scope, err := sm.getCurrentScope()
 	if err != nil {
 		return nil, err
 	}
 
 	if value, ok := (*scope)[name]; ok {
-		return value, nil
+		return &value, nil
 	}
 
 	return nil, fmt.Errorf("variable %s not found in current scope", name)
 }
 
-func (sm *ScopeManager) GetVariableValueAsBigInt(name string) (*big.Int, error) {
-	value, err := sm.GetVariableValue(name)
+func (sm *ScopeManager) GetVariableValueAsFelt(name string) (*fp.Element, error) {
+	value, err := sm.GetScopeValue(name)
 	if err != nil {
 		return nil, err
 	}
 
-	valueBig, ok := value.(*big.Int)
-	if !ok {
-		return nil, fmt.Errorf("value: %s is not a *big.Int", value)
+	return &value.Felt, nil
+}
+
+func (sm *ScopeManager) GetVariableValueAsBigInt(name string) (*big.Int, error) {
```

Comment:
same here

Reply:
Hey, what could I do here? I cant return nil if big.Int isn't a pointer

Reply:
Everything is working now